eth/catalyst/api.go

```diff
@@ -216,25 +220,23 @@ func (api *ConsensusAPI) ForkchoiceUpdatedV1(heads ForkchoiceStateV1, PayloadAtt
 		if err != nil {
 			return INVALID, err
 		}
-		hash := computePayloadId(heads.HeadBlockHash, PayloadAttributes)
-		id := binary.BigEndian.Uint64(hash)
-		api.preparedBlocks[id] = data
-		log.Info("Created payload", "payloadid", id)
-		// TODO (MariusVanDerWijden) do something with the payloadID?
-		hex := hexutil.Bytes(hash)
-		return ForkChoiceResponse{Status: SUCCESS.Status, PayloadID: &hex}, nil
+		id := computePayloadId(heads.HeadBlockHash, PayloadAttributes)
+		api.preparedBlocks.Add(id, data)
+		log.Info("Created payload", "payloadID", id)
+		return ForkChoiceResponse{Status: SUCCESS.Status, PayloadID: &id}, nil
 	}
 	return ForkChoiceResponse{Status: SUCCESS.Status, PayloadID: nil}, nil
 }
 
-func computePayloadId(headBlockHash common.Hash, params *PayloadAttributesV1) []byte {
+func computePayloadId(headBlockHash common.Hash, params *PayloadAttributesV1) (out PayloadID) {
```

Comment:
Please don't use naked returns ever and try not to use named returned either unless there are many returns with shared types. Naked returns make the code extremely hard to digest and are actually a deprecated feature of Go that they often said was a mistake but don't want to remove due to compatibility.

Reply:
When there's a single return output named `out`, I'm not as opposed. If you dislike it enough I would add it to linting so you don't have to catch it and we can fix it right away :)

Reply:
Changed it to not do the naked return. Go 1.17 allows slice to array conversion, but avoided that because of the minimum Go version in the repo.